### DISCUSSION TITLE
Check if a mutation result is optimistic

### DISCUSSION THREAD

**Discussion by chrisgco** [urql-graphql/urql#3436] [Q&A]
Hey is there a way to check if a mutation result is optimistic or if the result is from the server? It looks like `isFetching` is false when an optimistic result is returned so I can't figure out if I'm getting the real result.

I'm trying to achieve something like the following:

```tsx

const [{ result, fetching, error, isOptimistic }, executeMutation] = useMutation(MutationDocument)

return (
  <Button
     onPress={() => executeMutation()}
     isDisabled={isOptimistic || isFetching}
  />
)
```

So that multiple requests are not sent while the server request is loading.

**Comment by JoviDeCroock** [comment]
What I have done in the past is add a prefix/... to the `id` or another property to indicate that this is an optimistic response. I also think that `operation.context` should be carrying an `optimistic` flag.
